src/verify_cert.rs

```diff
@@ -295,6 +284,37 @@ fn check_signed_chain(
     Ok(())
 }
 
+fn check_signed_chain_name_constraints(
+    cert_chain: &Cert,
+    trust_anchor: &TrustAnchor,
+    subject_common_name_contents: subject_name::SubjectCommonNameContents,
+    budget: &mut Budget,
+) -> Result<(), Error> {
+    let mut cert = cert_chain;
```

Comment:
This seems weird? I assume we can rename the argument to `cert`?

Reply:
This is a bit of an [existing pattern](https://github.com/rustls/webpki/blob/99b64c57884e5a8e9b4b6f262eb3cc53c7b443d6/src/verify_cert.rs#L253-L261). The `Cert` type acts as a node in a linked list. When we pass in the `Cert` to the function argument it's standing-in as the head of the built chain. On L293 we're pulling out a mutable binding for the individual `Cert` being processed by the loop. We could update the fn signature to use `mut cert: &Cert` but that feels a little bit more confusing to me than the existing pattern. WDYT?

Reply:
The experiment sounds interesting. Have you come up with good names?

Reply:
> The experiment sounds interesting. Have you come up with good names?

I'll have to dig up an old branch. I was experimenting with this while trying to expose the built path back to the caller. At the time one of the big issues I was running into was not being able to return a reference to the [potential_issuer](https://github.com/rustls/webpki/blob/7e38e97cd1ae27a7ff2a8d48a2309db9f26d6dbd/src/verify_cert.rs#L216-L217) constructed inside the recursion that would outlast the frame. I thought that constructing `Cert` representations of all the intermediates in the caller, and only constructing the linked list nodes in the recursion would help. It didn't for reasons I can't remember now so I tabled the whole approach. Might make sense to revisit anyway

Reply:
I'm going to leave the name as-is (unless you feel strongly) and revisit the type split once I have a bit of spare time.